test/integration/consul-container/libs/consul-node/consul-container.go

```diff
@@ -0,0 +1,130 @@
+package consul_node
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/docker/docker/pkg/ioutils"
+
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/hashicorp/consul/api"
+)
+
+const bootLogLine = "Consul agent running"
+const disableRYUKEnv = "TESTCONTAINERS_RYUK_DISABLED"
+
+// consulContainerNode implement a ConsulNode
+// it instantiate Consul as a container
+type consulContainerNode struct {
+	ctx       context.Context
+	client    *api.Client
+	container testcontainers.Container
+	ip        string
+	port      int
+}
+
+// NewConsulContainer create a ConsulNode implemented as a consulContainerNode
+func NewConsulContainer(ctx context.Context, config Config) (ConsulNode, error) {
+
+	name := utils.RandName("consul-")
+	ctx = context.WithValue(ctx, "name", name)
```

Comment:
I couldn't find where this context value gets read? Is it inside testcontainers?

Reply:
it was a left over of an experiment I was doing. I will remove it